src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/CSharpCodeParser.cs

```diff
@@ -1556,6 +1557,25 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
                                 return;
                             }
 
+                            break;
+
+                        case DirectiveTokenKind.IdentifierOrExpression:
+                            if (At(SyntaxKind.Transition) && NextIs(SyntaxKind.LeftParenthesis))
+                            {
+                                EatCurrentToken();
+                                var expression = ParseExplicitExpressionBody();
+                                directiveBuilder.Add(expression);
+                            }
+                            else if (!TryParseQualifiedIdentifier(out identifierLength))
+                            {
+                                Context.ErrorSink.OnError(
+                                    RazorDiagnosticFactory.CreateParsing_DirectiveExpectsIdentifierOrExpression(
+                                        new SourceSpan(CurrentStart, identifierLength), descriptor.Directive));
+
+                                builder.Add(BuildDirective());
```

Comment:
Need to merge in main, this conflicts with https://github.com/dotnet/razor/pull/9195

Reply:
This is going into a branch right now. I'll do a merge from main into that branch and fix up any conflicts before we merge back into main.